src/components/kubectl/kubeconfig.ts

```diff
@@ -2,21 +2,60 @@ import * as vscode from 'vscode';
 import { fs } from '../../fs';
 import * as path from 'path';
 import * as yaml from 'js-yaml';
+import * as shelljs from 'shelljs';
+import { ConfigurationV1 } from '../../api/contract/configuration/v1';
 import { refreshExplorer } from '../clusterprovider/common/explorer';
-import { getActiveKubeconfig } from '../config/config';
+import { getActiveKubeconfig, getUseWsl } from '../config/config';
 
 interface Named {
     readonly name: string;
 }
 
+export function getKubeconfigPath(): ConfigurationV1.KubeconfigPath  {
```

Comment:
I feel like internal functions shouldn't be bound to API signatures.  I'd prefer `kubeconfig.ts` to define a canonical internal representation, and the API surfaces that private representation that (either, as in this case, directly via a compatible interface definition, or indirectly via an adapter).  Until and unless we can make the API so foundational that _even the core extension_ can be built out of the API, I feel that this is a mixing of layers: currently the API sits over the top of the internal code, rather than the internal code interoperating via the API, if that makes sense.

Reply:
I don't feel super strongly about this, and there is absolutely no harmful impact of leaving the code as you have it, so feel free to WONTFIX this comment!